### PR TITLE
fix: wandb config not saved in offline mode

### LINKED ISSUE #3607: Tracking with `WANDB_MODE="offline"` will not pass through the config dict
### System Info

```Shell
- `Accelerate` version: 1.7.0
- Platform: Linux-5.10.0-35-amd64-x86_64-with-glibc2.35
- `accelerate` bash location: /home/sdauncey/.local/bin/accelerate
- Python version: 3.11.12
- Numpy version: 2.2.5
- PyTorch version: 2.7.0+cu126
- PyTorch accelerator: CUDA
- System RAM: 2015.71 GB
- GPU type: NVIDIA A100 80GB PCIe
- `Accelerate` default config:
        Not found
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

1. Set the environment variable `WANDB_MODE` to `"offline"`:
```
WANDB_MODE="offline"

export WANDB_MODE

python debug_wandb_accelerate.py
```
2. Example python script `debug_wandb_accelerate.py
` (from [wandb website](https://docs.wandb.ai/guides/integrations/accelerate/)):
```
from accelerate import Accelerator

# Tell the Accelerator object to log with wandb
accelerator = Accelerator(log_with="wandb")

# Initialise your wandb run, passing wandb parameters and any config information
accelerator.init_trackers(
        project_name="debug_wandb",
        config={"dropout": 0.1, "learning_rate": 1e-2}
    )

for i in range(10):
    # Log to wandb by calling `accelerator.log`, `step` is optional
    accelerator.log({"train_loss": 1.12**(-i), "valid_loss": 0.8**(-i)}, step=i)


# Make sure that the wandb tracker finishes correctly
accelerator.end_training()
```
3. run `wandb sync /path/suggested/by/output`.
4. At the url from the output at `Syncing: https://wandb.ai/...` you _will_ find the logged losses but _will not_ find the `config` hyperparameters
Image
6. The variables `config={"dropout": 0.1, "learning_rate": 1e-2}` are also not logged in the `wandb-summary.json` file
7. Running the script without `WANDB_MODE="offline"` displays the config (see below image)
8. Running an adapted script to only use `wandb` and not `accelerate` displays the config (see below image)

Image

### Expected behavior

I expect that running with the environment variable `WANDB_MODE="offline"` would record the dictionary passed into `config` and display this with the rest of the data logged by `wandb`, which is the same behaviour as when this environment variable is not set or when `accelerate` is not used. 

**Workaround for now**: the config is recorded and displayed if you also pass the config dictionary into the `init_kwargs` variable that you pass to `init_trackers`:

eg. passing ` init_kwargs={"wandb": {"config": {"dropout": "abc"}}}` produces:

Image

### PR DESCRIPTION
**Pull request by ved1beta**
Pass config during wandb.init() in offline mode instead of using
config.update() 
Fixes #3607 
similar [https://github.com/huggingface/transformers/issues/38968](https://github.com/huggingface/transformers/issues/38968)

## Before submitting
- [X] Was this discussed/approved via a Github issue or the forum? Please add a link
      to it if that's the case.

## Who can review?
@SunMarc

### CODE REVIEW

**Comment by SunMarc** [comment]
can you check if this fixes your issue @SamD770 ?

**Comment by SunMarc** [comment]
@bot /style
